mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/UserReindexingTaskAdditionalInformationDTO.java

```diff
@@ -0,0 +1,77 @@
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.mailbox.tools.indexer;
+
+import java.util.List;
+import java.util.function.Function;
+
+import org.apache.james.core.User;
+import org.apache.james.json.DTOModule;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.server.task.json.dto.AdditionalInformationDTO;
+import org.apache.james.server.task.json.dto.AdditionalInformationDTOModule;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class UserReindexingTaskAdditionalInformationDTO implements AdditionalInformationDTO {
+
+    public static final Function<MailboxId.Factory, AdditionalInformationDTOModule<UserReindexingTask.AdditionalInformation, UserReindexingTaskAdditionalInformationDTO>> SERIALIZATION_MODULE =
+        factory ->
+            DTOModule.forDomainObject(UserReindexingTask.AdditionalInformation.class)
+                .convertToDTO(UserReindexingTaskAdditionalInformationDTO.class)
+                .toDomainObjectConverter(dto -> new UserReindexingTask.AdditionalInformation(User.fromUsername(dto.getUser()),
+                    dto.getSuccessfullyReprocessedMailCount(),
+                    dto.getFailedReprocessedMailCount(),
+                    ReprocessingContextInformationDTO.deserializeFailures(factory, dto.getFailures())))
+                .toDTOConverter((details, type) -> new UserReindexingTaskAdditionalInformationDTO(details.getUser(), details.getSuccessfullyReprocessedMailCount(), details.getFailedReprocessedMailCount(),
+                    ReprocessingContextInformationDTO.serializeFailures(details.failures())))
+                .typeName(UserReindexingTask.USER_RE_INDEXING.asString())
+                .withFactory(AdditionalInformationDTOModule::new);
+
+    private final ReprocessingContextInformationDTO reprocessingContextInformationDTO;
+    private final String user;
+
+    @JsonCreator
+    private UserReindexingTaskAdditionalInformationDTO(@JsonProperty("user") String user,
+                                                       @JsonProperty("successfullyReprocessedMailCount") int successfullyReprocessedMailCount,
+                                                       @JsonProperty("failedReprocessedMailCount") int failedReprocessedMailCount,
+                                                       @JsonProperty("failures") List<ReprocessingContextInformationDTO.ReindexingFailureDTO> failures) {
+        this.user = user;
+        this.reprocessingContextInformationDTO = new ReprocessingContextInformationDTO(
```

Comment:
IMO here using `ReprocessingContextInformationDTO` just makes things less clear (and we do not return this type at all).

Can't we just have `int` + `List<ReprocessingContextInformationDTO.ReindexingFailureDTO>` to avoid logic in constructor and getter ?

Reply:
I think that the encapsulation of the `ReprocessingContextInformationDTO` make the code more maintainable in case we add a field in it. The compiler will force us to update all the DTOs using it.